tests/Fixture/TestParticle.ts

```diff
@@ -13,6 +13,29 @@ export class TestParticle {
         this.particle = new Particle(this.container, this.position);
     }
 
+    /**
+     * When creating a particle, the private method `Particle.calcPosition` ensures
+     * the particle is completely inside the canvas. It does this by shifting the
+     * position if any part of the particle is outsize the canvas. This function
+     * returns random positions of particles that are guaranteed to lie completely
+     * inside the canvas.
+     *
+     * @param container
+     */
+    public randomPositionInCanvas(container?: Container): ICoordinates {
```

Comment:
This behavior will probably change in the future, I was thinking about this positioning. For now it's fine if it works with the current version.